Add pre_update_term filter hook

2mkabir opened the pull request:
Hi
In file:
`wp-includes/taxonomy.php`
In function:
`wp_update_term`
How can to validate a custom term before updated?
A filter needs here like `pre_insert_term` in `wp_insert_term()` functin.
I think this filter has been forgotten in here.
`$term = apply_filters( 'pre_update_term', $term, $taxonomy, $args );`

https://core.trac.wordpress.org/ticket/58404

<!--
Hi there! Thanks for contributing to WordPress!

Pull Requests in this GitHub repository **must** be linked to a ticket in the WordPress Core Trac instance (https://core.trac.wordpress.org), and are only used for code review. **No pull requests will be merged on GitHub.**

See the WordPress Handbook page on using PRs for Code Review more information: https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/

If this is your first time contributing, you may also find reviewing these guides first to be helpful:
- FAQs for New Contributors: https://make.wordpress.org/core/handbook/tutorials/faq-for-new-contributors/
- Contributing with Code Guide: https://make.wordpress.org/core/handbook/contribute/
- WordPress Coding Standards: https://make.wordpress.org/core/handbook/best-practices/coding-standards/
- Inline Documentation Standards: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/
- Browser Support Policies: https://make.wordpress.org/core/handbook/best-practices/browser-support/
- Proper spelling and grammar related best practices: https://make.wordpress.org/core/handbook/best-practices/spelling/
-->

<!-- Insert a description of your changes here -->

Trac ticket: <!-- insert a link to the WordPress Trac ticket here -->

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

2mkabir committed:
Add pre_update_term filter hook
https://core.trac.wordpress.org/ticket/58404

joemcgill commented:
Sorry for the mistaken review request. Looking at this, I'm unsure what the filter would provide that the current `wp_update_term_data` filter does not. Could you provide an example use case where this would be useful, @2mkabir?

2mkabir commented:
Hi @joemcgill. I need `pre_update_term` filter like this:

```
$term = apply_filters( 'pre_update_term', $term, $taxonomy, $args );

if ( is_wp_error( $term ) ) {
     return $term;
}
```

By utilizing this filter, I can supply a custom `WP_Error()` object as the return value for `$term`. This is analogous to the behavior of the `pre_insert_term` filter.

It serves its purpose within the `/wp-admin/edit-tags.php` page, specifically in the validation process of the custom field within the "quick edit" form.

### About my use case

I have created a plugin for Woocommerce [select-quantity-by-attribute-for-woocommerce](https://github.com/2mkabir/select-quantity-by-attribute-for-woocommerce)
I have added a custom field to the configure terms page of the product attributes. A sample url of this page:
`/wp-admin/edit-tags.php?taxonomy=pa_quantity&post_type=product`
This custom field is require and also added to the quick edit form.
My problem is that there is no way for validation my custom require field.

My custom field (Quantity). I want to show the error message if the Quantity field be empty
![](https://i.ibb.co/bLSkwVC/Screenshot-2023-09-26-014130.png)

Show error message for name field that is default of the Wordpress.
I want to do this work for the Quantity field.
![](https://i.ibb.co/DD8tpL5/Screenshot-2023-09-26-014342.png)

This is my code that doing validation with `pre_insert_term` filter.  Also I want to do like this with `pre_update_term` filter. 
https://github.com/2mkabir/select-quantity-by-attribute-for-woocommerce/blob/master/select-quantity-by-attribute-for-woocommerce.php#L99
